.travis.yml

```diff
@@ -1,12 +1,50 @@
 language: python
-sudo: enabled
+sudo: required
+dist: trusty
+addons:
+  ssh_known_hosts: localhost
+  postgresql: "9.5"
+cache:
+  pip: true
+  directories:
+    - assembl/static/js/node_modules
+    - assembl/static/widget/card/bower_components
+    - assembl/static/widget/creativity/bower_components
+    - assembl/static/widget/session/bower_components
+    - assembl/static/widget/share/bower_components
+    - assembl/static/widget/video/bower_components
+    - assembl/static/widget/vote/bower_components
+    - assembl/static2/node_modules
 before_install:
- - sudo apt-get -qq update
+ # The following line is needed, otherwise the set_file_permissions fab command yells error "Operation not permitted"
+ - sudo usermod -a -G www-data $USER
  - sudo apt-get install -y fabric git openssh-server
- - fab -c configs/develop.rc install_single_server
- - fab -c configs/develop.rc bootstrap_from_checkout
-install: true
+ - ssh-keygen -P '' -f ~/.ssh/id_rsa && cat ~/.ssh/id_rsa.pub >> ~/.ssh/authorized_keys
+ - echo "_extends = ./testing.rc" > configs/travis.rc; echo "_gitbranch = $TRAVIS_COMMIT" >> configs/travis.rc;
+ # The following line produces warning "uses weak digest algorithm" during its `apt-get update`
+ - fab -c configs/travis.rc install_assembl_server_deps
+ - fab -c configs/travis.rc install_redis
+ - fab -c configs/travis.rc install_memcached
+install:
+ # The set_file_permissions fabric command needs to be encapsulated into this sudo su command, because in before_install we have added $USER in www-data group but have not logged out and logged in again for changes to take effect in this terminal
+ - sudo -E su $USER -c 'fab -c configs/travis.rc set_file_permissions'
+ # TODO: fix fabfile's check_and_create_database_user command which produces an error when run by travis. For now we instead run the following command
+ - pip install psycopg2
+ - python2 $PWD/assembl/scripts/pypsql.py -u postgres -n localhost  "CREATE USER assembl_test WITH CREATEDB ENCRYPTED PASSWORD 'assembl_test'; COMMIT;"
```

Comment:
ooh boy, this isn't good. Ahh, is it failing because the script `assembl-pypsql` doesn't exist yet?
If so, prior to running this, you can run:
```
pip install -e $PWD
```

And that would install the setup.py scripts in "editable mode", giving you access to them. 

https://pip.pypa.io/en/stable/reference/pip_install/#editable-installs

Reply:
Maybe could resolve this issue.

Reply:
That may not be necessary. We install psycopg2 globally so the script can be run before the whole assembl machinery. `python assembl/scripts/pypsql.py ...` should work.